src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -438,13 +439,20 @@ export function generateSignedTransactionForSimulation(args: InputSimulateTransa
     );
   } else if (accountAuthenticator instanceof AccountAuthenticatorSingleKey) {
     transactionAuthenticator = new TransactionAuthenticatorSingleSender(accountAuthenticator);
+  } else if (accountAuthenticator instanceof AccountAuthenticatorNoAccountAuthenticator) {
+    transactionAuthenticator = new TransactionAuthenticatorSingleSender(accountAuthenticator);
   } else {
     throw new Error("Invalid public key");
   }
   return new SignedTransaction(transaction.rawTransaction, transactionAuthenticator).bcsToBytes();
 }
 
 export function getAuthenticatorForSimulation(publicKey: PublicKey) {
+  // TODO: check this.
+  if (publicKey.toString() === (new Ed25519PublicKey(new Uint8Array(32))).toString()) {
```

Comment:
what are you trying to check here? It feels error-prone.

Reply:
I updated this part so that `null` (rather than `000...0`) represents the absence of a public key.

Reply:
Update this.